okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.java

```diff
@@ -99,6 +106,12 @@
   /** True if we've enqueued a close frame. No further message frames will be enqueued. */
   private boolean enqueuedClose;
 
+  /**
+   * When executed this will cancel this websocket. This future itself should be canceled if that is
+   * unnecessary because the web socket is already closed or canceled.
+   */
+  private ScheduledFuture<?> cancelFuture;
```

Comment:
Need to keep this so that we can release the ScheduledExecutorService thread as eagerly as possible.